specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/compute.json

```diff
@@ -0,0 +1,15112 @@
+          },
+          {
+            "name": "sshPublicKeyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the SSH public key."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyUpdateResource"
+            },
+            "description": "Parameters supplied to update the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyResource"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "SshPublicKeys"
+        ],
+        "operationId": "SshPublicKeys_Delete",
+        "description": "Delete an SSH public key.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "sshPublicKeyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "SshPublicKeys"
+        ],
+        "operationId": "SshPublicKeys_Get",
+        "description": "Retrieves information about an SSH public key.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "sshPublicKeyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyResource"
```

Comment:
Can we rename this model to `SshPublicKey` to align the name of the operation group? We actually do not need to emphasize the `Resource` keyword on a resource model.

Reply:
@ArcturusZhang 

this has been around for a couple versions now. I belive they named it this way, because there is a property named "SshPublicKey" within virtual machine already, and this is a azure resource. 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L9312

Reply:
Yeah - if we change this, we will have to change the `SshPublicKey` within virtual machine too.
Is it possible to change both of them? For instance, I believe it makes more sense if we rename the `SshPublicKey` within virtual machine to `SshPublicKeyConfiguration`, and then we can rename the `SshPublicKeyResource` to `SshPublicKey`